### PR TITLE
fix for #718, #714, #711, #710

### PR DESCRIPTION
**Pull request by bhagyashree-sarawate**
Hi 
Fixes for #718,#714, #711 
Please review
Regards
Bhagyashree Sarawate

### CODE REVIEW

**Comment by bhagyashree-sarawate** [comment]
Added the fix for#710
Please review

**Comment by sneharai4** [comment]
@bhagyashree-sarawate Code looks good to me.

File: hpedockerplugin/volume_manager.py [hunk 1]
```diff
@@ -119,9 +119,10 @@ def _initialize_configuration(self):
                         "Failed to initialize driver - cpg_map not defined for"
                         "replication device")
 
-                self.tgt_bkend_config.hpe3par_snapcpg = \
-                    self._extract_remote_cpgs(
-                        self.tgt_bkend_config.snap_cpg_map)
+                if self.tgt_bkend_config.snap_cpg_map:
```
Comment (sneharai4) [inline hpedockerplugin/volume_manager.py]: @bhagyashree-sarawate As per https://github.com/hpe-storage/python-hpedockerplugin/blob/master/docs/active-passive-based-replication.md both _cpg_map_ and _snap_cpg_map_ in replication_device section are mandatory. 

But as per code, when user doesn't give _snap_cpg_map_, cpg_map is used as snap_cpg_map.

Please fix the documentation for this.

Comment (bhagyashree-sarawate) [reply hpedockerplugin/volume_manager.py]: done

File: docs/active-passive-based-replication.md [hunk 2]
```diff
@@ -40,7 +40,7 @@ replication_device = backend_id:<Target-Array-Hostname>,
 1. In case of asynchronous replication mode, *sync_period* field can optionally be 
 defined as part of *replication_device* entry and it should be between range 300 
 and 31622400 seconds. If not defined, it defaults to 900 seconds.
-2. Both *cpg_map* and *snap_cpg_map* in *replication_device* section are mandatory.
+2. *cpg_map* and *snap_cpg_map* in *replication_device* section are mandatory. If *snap_cpg_map* is not mentioned then it will be same as *cpg_map*
```
Comment (sneharai4) [inline docs/active-passive-based-replication.md]: @bhagyashree-sarawate Please remove the space at the end here.

Comment (bhagyashree-sarawate) [reply docs/active-passive-based-replication.md]: done

**Pull request merged**